### PR TITLE
Revisions for support of binary literals

### LINKED ISSUE #1709: Support for binary value assignments (e.g., x = 0b01;)
The documentation should note how to create numeric values not only from decimal literals, but also from hexadecimal and (starting in C# 7) binary literals.

### PR DESCRIPTION
**Pull request by rpetrusha**
# Revisions for support of binary literals

## Summary

Documents support for 0x to define hexadecimal literals and 0b to define binary literals

Fixes #1709

## Details

Expands the literal assignment section of each primitive integral type (`byte`, `short`, `int`, `long`, `sbyte`, `ushort`, `uint`, `ulong` to include hex and binary assignment.

Also fixed a number of other issues, including absence of language in fenced code blocks, organization of section, separating literals from compiler overload resolution, etc.

## Suggested Reviewers

@BillWagner

### CODE REVIEW

File: docs/csharp/language-reference/keywords/long.md [hunk 1]
```diff
@@ -32,60 +32,73 @@ translation.priority.ht:
...
   
- When you use the suffix L, the type of the literal integer is determined to be either `long` or [ulong](../../../csharp/language-reference/keywords/ulong.md) according to its size. In the case it is `long` because it less than the range of [ulong](../../../csharp/language-reference/keywords/ulong.md).  
+ When you use the suffix `L`, the type of the literal integer is determined to be either `long` or [ulong](../../../csharp/language-reference/keywords/ulong.md), depending on its size. In this case, it is `long` because it less than the range of [ulong](../../../csharp/language-reference/keywords/ulong.md).  
   
- A common use of the suffix is with calling overloaded methods. Consider, for example, the following overloaded methods that use `long` and [int](../../../csharp/language-reference/keywords/int.md) parameters:  
+ A common use of the suffix is to call overloaded methods. Tor example, the following overloaded methods have parameters of type `long` and [int](../../../csharp/language-reference/keywords/int.md):
```
Comment (BillWagner) [inline docs/csharp/language-reference/keywords/long.md]: typo: Tor => For

File: samples/snippets/csharp/language-reference/keywords/numeric-literals.cs [hunk 2]
```diff
@@ -0,0 +1,314 @@
...
+   private static void AssignShort()
+   {
+      // <SnippetShort>
+      short shortValue1 = 1034;
+      Console.WriteLine(shortValue1);
+      
+      short shortValue2 = 0x040A;
+      Console.WriteLine(shortValue2);
+      
+      short shortValue3 = 0b010000001010;
```
Comment (BillWagner) [inline samples/snippets/csharp/language-reference/keywords/numeric-literals.cs]: I'd recommend adding the _ separator in all the binary constants. It does improve readability

**Review by BillWagner — CHANGES_REQUESTED** [review]
@rpetrusha overall, this looks great. I found one typo, and I had one suggestion. Once that's done, let's :shipit:

File: docs/csharp/language-reference/keywords/long.md (hunk 1 above)
Comment (rpetrusha) [inline docs/csharp/language-reference/keywords/long.md]: Fixed.

File: samples/snippets/csharp/language-reference/keywords/numeric-literals.cs (hunk 2 above)
Comment (rpetrusha) [inline samples/snippets/csharp/language-reference/keywords/numeric-literals.cs]: Definitely. Done.

File: docs/csharp/language-reference/keywords/uint.md [hunk 3]
```diff
@@ -41,46 +42,41 @@ The `uint` keyword signifies an integral type that stores values according to th
...
+[!code-cs[uint](../../../../samples/snippets/csharp/language-reference/keywords/numeric-literals.cs#UInt)]  
+
+> [!NOTE] 
+> You use the prefix `0x` or `0X` to denote a hexadecimal literal and the prefix `0b` or `0B` to denote a binary literal. Decimal literals have no prefix. 
+
+Starting with C# 7, you can also use the underscore character, `_`, as a digit separator to enhance readability, as the following example shows.
+
+[!code-cs[uint](../../../../samples/snippets/csharp/language-reference/keywords/numeric-literals.cs#UIntS)]  
+ 
+ Integer literals can also include a suffix that denotes the type. The suffix `U` or 'u' denotes either a `uint` or a `ulong`, depending on the numeric value of the literal. The following example uses the `u` suffix to denote an unsigned integer of both types. Note that the first literal is an `uint` because its value is less than <xref:System.UInt32.MaxValue?displayProperty=fullName>, while the second is a `ulong` because its value is greater than <xref:System.UInt32.MaxValue?displayProperty=fullName>.
```
Comment (mairaw) [inline docs/csharp/language-reference/keywords/uint.md]: an `uint` -> a `uint`?

Comment (rpetrusha) [inline docs/csharp/language-reference/keywords/uint.md]: Fixed.

**Review by BillWagner — APPROVED** [review]

**Pull request merged**